internal/service/alarms/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	api "github.com/openshift-kni/oran-o2ims/internal/service/alarms/api/generated"
+	"github.com/openshift-kni/oran-o2ims/internal/service/alarms/internal"
+)
+
+func main() {
+	Serve()
+}
+
+// Serve TODO: Call this func using cobra-cli from inside deployment CR.
+func Serve() {
+	// TODO: Init client-go
+
+	// TODO: Init DB client
+
+	// TODO: Audit and Insert data database
+
+	// TODO: Launch k8s job for DB remove archived data
+
+	// Init server
+	r := http.NewServeMux()
+
+	// Register
+	alarmServerStrict := api.NewStrictHandlerWithOptions(internal.AlarmsServer{}, nil,
+		api.StrictHTTPServerOptions{
+			RequestErrorHandlerFunc:  getOranReqErrFunc(),
+			ResponseErrorHandlerFunc: getOranRespErrFunc(),
+		})
+
+	opt := api.StdHTTPServerOptions{
+		BaseRouter: r,
+		Middlewares: []api.MiddlewareFunc{ // Add middlewares here
+			internal.AlarmsOapiValidation(),
+			internal.LogDuration(),
+		},
+		ErrorHandlerFunc: getOranReqErrFunc(),
+	}
+
+	// Create the handler
+	handler := api.HandlerWithOptions(alarmServerStrict, opt)
+
+	// Server config
+	srv := &http.Server{
+		Handler:      handler,
+		Addr:         net.JoinHostPort("0.0.0.0", "8080"),
```

Comment:
we should restrict this to `127.0.0.1` since we have a proxy sitting in front of us.

Reply:
updated